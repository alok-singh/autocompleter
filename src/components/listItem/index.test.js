import React from "react";
import { render } from "@testing-library/react";
import ListItem from "./index";

const dummuyData = [
  {
    title: "Thor",
    year: "2011",
    imdbID: "tt0800369",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BOGE4NzU1YTAtNzA3Mi00ZTA2LTg2YmYtMDJmMThiMjlkYjg2XkEyXkFqcGdeQXVyNTgzMDMzMTg@._V1_SX300.jpg",
  },
  {
    title: "Thor: The Dark World",
    year: "2013",
    imdbID: "tt1981115",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BMTQyNzAwOTUxOF5BMl5BanBnXkFtZTcwMTE0OTc5OQ@@._V1_SX300.jpg",
  },
  {
    title: "Thor: Ragnarok",
    year: "2017",
    imdbID: "tt3501632",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BMjMyNDkzMzI1OF5BMl5BanBnXkFtZTgwODcxODg5MjI@._V1_SX300.jpg",
  },
  {
    title: "Team Thor",
    year: "2016",
    imdbID: "tt6016776",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BMmY4NzA2OGUtYTc0MS00OTc0LThhYjAtZTJkMTI2MTgyNDY0XkEyXkFqcGdeQXVyNDQ5MDYzMTk@._V1_SX300.jpg",
  },
  {
    title: "Thor: Tales of Asgard",
    year: "2011",
    imdbID: "tt1667903",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BMTcxOTU4NzIxMV5BMl5BanBnXkFtZTcwNzE3NjAxNQ@@._V1_SX300.jpg",
  },
  {
    title: "Team Thor: Part 2",
    year: "2017",
    imdbID: "tt6599818",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BOTJkYmFkM2ItOGIwMS00YWRhLTg5ZGQtNDA0ZjcwNzA3ZjQxXkEyXkFqcGdeQXVyNDQ5MDYzMTk@._V1_SX300.jpg",
  },
  {
    title: "Almighty Thor",
    year: "2011",
    imdbID: "tt1792794",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BMTcwNjI5MTAzNF5BMl5BanBnXkFtZTcwNTcyOTIwNQ@@._V1_SX300.jpg",
  },
  {
    title: "Thor: Hammer of the Gods",
    year: "2009",
    imdbID: "tt1260572",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BNTFkMDA5ODEtMTQ1NC00NGRhLWE3ZGQtOGE5ZGJkMGI1ZDI5XkEyXkFqcGdeQXVyMjI1NjEzNjg@._V1_SX300.jpg",
  },
  {
    title: "Thor: Legend of the Magical Hammer",
    year: "2011",
    imdbID: "tt1241721",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BMTYxOTI1OTI0NF5BMl5BanBnXkFtZTgwMDQzMjcwMzE@._V1_SX300.jpg",
  },
  {
    title: "I Am Thor",
    year: "2015",
    imdbID: "tt2450258",
    type: "movie",
    poster:
      "https://m.media-amazon.com/images/M/MV5BNDQxNTI4Mjg1MV5BMl5BanBnXkFtZTgwMDM2MTMyNzE@._V1_SX300.jpg",
  },
];

test("unit testing for list item in autocompleter list", () => {
  expect(
    render(
    <div>{dummuyData.map(item => <ListItem key={item.imdbID} {...item} />)}</div>
    ).baseElement
  ).toMatchSnapshot();
});
