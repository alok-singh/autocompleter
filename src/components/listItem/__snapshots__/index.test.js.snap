// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit testing for list item in autocompleter list 1`] = `
<body>
  <div>
    <div>
      <div
        class="list-item"
      >
        <img
          alt="Thor"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Thor
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2011
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Thor: The Dark World"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Thor: The Dark World
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2013
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Thor: Ragnarok"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Thor: Ragnarok
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2017
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Team Thor"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Team Thor
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2016
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Thor: Tales of Asgard"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Thor: Tales of Asgard
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2011
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Team Thor: Part 2"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Team Thor: Part 2
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2017
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Almighty Thor"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Almighty Thor
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2011
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Thor: Hammer of the Gods"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Thor: Hammer of the Gods
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2009
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="Thor: Legend of the Magical Hammer"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            Thor: Legend of the Magical Hammer
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2011
          </div>
        </div>
      </div>
      <div
        class="list-item"
      >
        <img
          alt="I Am Thor"
          class="u-image item-image"
        />
        <div
          class="item-description"
        >
          <div
            class="item-title"
          >
            I Am Thor
          </div>
          <div
            class="item-type"
          >
            movie
          </div>
          <div
            class="item-year"
          >
            2015
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
